### PR TITLE
Makes the Shield Capacitor UI into a TGUI.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Made the Shield Capacitor UI into a TGUI.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/modules/shieldgen/shield_capacitor.dm [hunk 1]
```diff
@@ -68,36 +68,40 @@
...
+	return data
+
+/obj/machinery/shield_capacitor/ui_act(action, list/params, datum/tgui/ui, datum/ui_state/state)
+	. = ..()
+	if(.)
+		return
+	switch(action)
+		if("toggle")
+			if(!active && !anchored)
+				to_chat(usr, SPAN_WARNING("The [src] needs to be firmly secured to the floor first."))
```
Comment (SleepyGemmy) [inline code/modules/shieldgen/shield_capacitor.dm:98]:
```suggestion
				to_chat(usr, SPAN_WARNING("\The [src] needs to be firmly secured to the floor first."))
```

**Review by SleepyGemmy — APPROVED** [review]

File: code/modules/shieldgen/shield_capacitor.dm [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 	desc = "Machine that charges a shield generator."
 	icon = 'icons/obj/machinery/shielding.dmi'
 	icon_state = "capacitor"
 	obj_flags = OBJ_FLAG_ROTATABLE
-	var/active = FALSE
 	density = TRUE
-	var/stored_charge = 0	//not to be confused with power cell charge, this is in Joules
+	/// Doesn't use APC power.
+	use_power = POWER_USE_OFF
+	req_one_access = list(access_captain, access_security, access_engine)
```
Comment (DreamySkrell) [inline code/modules/shieldgen/shield_capacitor.dm:10]: Why `access_security`?

Comment (NonQueueingMatt) [reply code/modules/shieldgen/shield_capacitor.dm:10]: its how it was before

**Pull request merged**